### PR TITLE
moving uiChanges fixes to new branch

### PR DESCRIPTION
**Pull request by mzieg**
Made some tweaks to uiChanges, which I unwisely committed directly to uiChanges, complicating the ability to do a Pull Request and show what I changed.  So I tried pulling them out and putting them back on a fresh branch, but I'm not sure I've done that correctly :-(

### CODE REVIEW

File: enlighten/Controller.py [hunk 1]
```diff
@@ -2462,38 +2440,19 @@ def generate_x_axis(self, spec=None, settings=None, unit=None, vignetted=True):
...
-        if self.roi_enabled:
-            self.form.ui.pushButton_roi_toggle.setStyleSheet("background-color: #aa0000")
-        else:
-            self.form.ui.pushButton_roi_toggle.setStyleSheet(self.default_roi_btn)
-        for spec in self.multispec.get_spectrometers():
-            self.graph.update_roi_regions(spec)
-
-    def get_roi_enabled(self):
-        return self.roi_enabled
-
```
Comment (mzieg) [inline enlighten/Controller.py:2496]: moved to VignetteROIFeature

File: enlighten/Controller.py [hunk 2]
```diff
@@ -2553,32 +2512,6 @@ def open_log(self):
...
-            else:
-                None
-
-        for name, obj_path in self.enlighten_graphs.items():
-            plot = functools.reduce(resolve_graph_plot, obj_path)
-            if plot != None:
-                plot.showGrid(self.grid_display, self.grid_display)
-            else:
-                log.error(f"{name} couldn't set grid")
-
```
Comment (mzieg) [inline enlighten/Controller.py:2581]: moved to GridFeature

File: enlighten/Graph.py [hunk 3]
```diff
@@ -492,52 +499,41 @@ def update_roi_regions(self, spec):
...
-            spectrum_end_cm = utils.wavelength_to_wavenumber(spec.settings.eeprom.excitation_nm, spectrum_end_nm)
+        log.debug(f"update_roi_regions: roi {roi}, axis {len(axis)} elements")
 
-            roi_start_cm = utils.wavelength_to_wavenumber(spec.settings.eeprom.excitation_nm_float, roi_start_nm)
-            roi_end_cm = utils.wavelength_to_wavenumber(spec.settings.eeprom.excitation_nm_float, roi_end_nm)
+        spec.roi_region_left .setRegion((axis[0],       axis[roi.start]))
+        spec.roi_region_right.setRegion((axis[roi.end], axis[-1]       ))
 
-            spec.roi_region_left.setRegion((-1*spectrum_start_cm, -1*roi_start_cm))
-            spec.roi_region_right.setRegion((-1*roi_end_cm, -1*spectrum_end_cm))
```
Comment (mzieg) [inline enlighten/Graph.py:527]: I never did figure out why you had to do those -1, but the new version is simpler

File: enlighten/Marquee.py [hunk 4]
```diff
@@ -36,7 +36,7 @@ def __init__(self,
         self.stylesheets = stylesheets
 
         self.original_height = self.frame.height()
-        self.original_height = 50 
+        self.original_height = 36
```
Comment (mzieg) [inline enlighten/Marquee.py:39]: honestly curious why we have to hardcode this.

File: enlighten/Measurement.py [hunk 5]
```diff
@@ -569,7 +569,7 @@ def update_label(self, label, manual=False):
         if self.thumbnail_widget:
             self.thumbnail_widget.rename(label)
             was_displayed = self.thumbnail_widget.is_displayed
-            self.thumbnail_widget.remove_curve_from_graph(label=old_label) # remove using the old value
+            self.thumbnail_widget.remove_curve_from_graph()
```
Comment (mzieg) [inline enlighten/Measurement.py:572]: this was buggy when I had multiple traces all called "400ms 8dB"

File: enlighten/Multispec.py [hunk 6]
```diff
@@ -89,6 +87,7 @@ def __init__(self,
         self.in_process = {} # a dict of device_id to WasatchDeviceWrapper (may be None if "gave up")
         self.disconnecting = {}
         self.ignore = {}
+        self.seen_serial_numbers = {} # dict of serial_number to...not sure yet.  Should include initial assigned color
```
Comment (mzieg) [inline enlighten/Multispec.py:90]: Might use this later to persist the first color code assigned to each serial number across a session

File: enlighten/ThumbnailWidget.py [hunk 7]
```diff
@@ -421,14 +425,14 @@ def update_kia(self):
...
-        if label is None:
-            label = self.le_name.text()
-        if self.graph.remove_curve(label):
-            self.set_active(False)
-            self.curve = None
-            return True
-        return False
+    def remove_curve_from_graph(self):
+        if self.measurement is not None:
+            self.graph.remove_curve(measurement_id=self.measurement.measurement_id)
```
Comment (mzieg) [inline enlighten/ThumbnailWidget.py:430]: main fix to bug involving duplicate labels

**Review by rc14193 — APPROVED** [review]

**Pull request merged**